.github/workflows/build-docker-image.yml

```diff
@@ -71,15 +71,16 @@ jobs:
           # list of Docker images to use as base name for tags
           images: |
             us-docker.pkg.dev/zealous-zebra/zebra/${{ inputs.image_name }}
+            zfnd/zebra,enable=${{ github.event_name == 'release' && !github.event.release.prerelease }}
           # generate Docker tags based on the following events/attributes
           tags: |
```

Comment:
Can you show me the full set of tags for each release?
And can you document them somewhere?

in particular, I am want to check these things:
- the version tag includes "-beta.15" or "-rc.0"
- the "sha" is long enough to be unique (git currently recommends 9 characters for our repository, and it will grow over time as we get more commits)
- we're using "v1" but "1.0", this seems inconsistent and a bit confusing?
- what if the ref looks like a version?

Reply:
> * the version tag includes "-beta.15" or "-rc.0"

These are tags for a `non-beta` or `non-rc` release:
https://github.com/ZcashFoundation/zebra/actions/runs/3053723143/jobs/4924709856#step:4:98

These are the tags for a `-beta.15` or `-rc.0`
https://github.com/ZcashFoundation/zebra/actions/runs/3053583083/jobs/4924390909#step:4:95

When a release specifies a `pre-release identifier` it just add the specific version tag and sha, it does not add `latest`, `v{{major}}`, `{{major}}.{{minor}}`, and neither of the other ones.

> * the "sha" is long enough to be unique (git currently recommends 9 characters for our repository, and it will grow over time as we get more commits)

This is automatically generated by https://github.com/docker/metadata-action#typesha and if we'd like to use the long version, this would need to be standardized on all the actions (this could be done later on)

> * we're using "v1" but "1.0", this seems inconsistent and a bit confusing?

If we have 1.0, 1.2, 1.3, and even another major version like 2.0, 2.2, 2,3, etc. We might want the latest version from version 1, this is `v1` as the `latest` would be 2.3. This is easier for the user to keep downloading the latest version from a specific major.

Rust has a good example in their [Docker Hub](https://hub.docker.com/_/rust), for example: `1-buster`, `1.63-buster`, `1.63.0-buster`, `buster`. If we remove the alias (`buster`) then we would have `1`, `1.63`, `1.63.0`

Reply:
> > * the "sha" is long enough to be unique (git currently recommends 9 characters for our repository, and it will grow over time as we get more commits)
> 
> This is automatically generated by https://github.com/docker/metadata-action#typesha and if we'd like to use the long version, this would need to be standardized on all the actions (this could be done later on)

It's not a blocker, but it might be confusing for users if we ever tag two releases with the same short sha. We can fix this later if it becomes a significant problem.

> > * we're using "v1" but "1.0", this seems inconsistent and a bit confusing?
> 
> If we have 1.0, 1.2, 1.3, and even another major version like 2.0, 2.2, 2,3, etc. We might want the latest version from version 1, this is `v1` as the `latest` would be 2.3. This is easier for the user to keep downloading the latest version from a specific major.
> 
> Rust has a good example in their [Docker Hub](https://hub.docker.com/_/rust), for example: `1-buster`, `1.63-buster`, `1.63.0-buster`, `buster`. If we remove the alias (`buster`) then we would have `1`, `1.63`, `1.63.0`
> 
> In Rust case `buster` is their major alias, in our case `v1` would be our alias.

Rust does not have `v` in front of any of its versions. Can we remove the extra `v`, for consistency with the Rust ecosystem, and consistency between versions?
(See my suggestion below.)